code/modules/events/ghost_role/operative.dm

```diff
@@ -32,6 +32,13 @@
 
 	Mind.add_antag_datum(/datum/antagonist/nukeop/lone)
 
+	// If we have a Station Trait "Naval Patrol" it will not prevent the Lone Ops but just announce it presence
+	// You should've secured that disk! Not even Navy will stop a disk-sniffing-determined Lone Operative
+	if(GLOB.dynamic_ruleset_categories == RULESET_CATEGORY_NO_OUTSIDE_ANTAGONISTS)
+		minor_announce(
+			message = "We've detected a fast-moving, human-sized, blood-red blip on radar heading towards [GLOB.station_name]. It somehow evaded our engagement protocols and is now within your proximity. The situation is yours to handle, no reply.",
```

Comment:
This is a neat detail -- can we get the same for the battlecruiser final objective? (not that they don't announce themselves explosively 5 minutes after spawn regardless)